dashbuilder/dashbuilder-backend/dashbuilder-services/src/main/java/org/dashbuilder/DashbuilderMigrationService.java

```diff
@@ -0,0 +1,154 @@
+        this.oldDatasetsFS = ioService.getOrNewFileSystem(URI.create("git://system/datasets"),
+                                                          new HashMap<String, Object>());
+        this.datasetsFS = datasetsFS;
+        this.pluginsFS = pluginsFS;
+        this.perspectivesFS = perspectivesFS;
+        this.navigationFS = navigationFS;
+    }
+
+    @PostConstruct
+    private void init() {
+        migrateDatasets(oldDatasetsFS, datasetsFS);
+        migratePerspectives(pluginsFS, perspectivesFS);
+        migrateNavigation(pluginsFS, navigationFS);
+    }
+
+    public void migrateDatasets(FileSystem sourceFS, FileSystem targetFS) {
+        migrate(sourceFS,
+                targetFS,
+                path -> true);
+    }
+
+    public void migratePerspectives(FileSystem sourceFS, FileSystem targetFS) {
+        migrate(sourceFS,
+                targetFS,
+                path -> path.getFileName().toString().startsWith("perspective_layout"));
+    }
+
+    public void migrateNavigation(FileSystem sourceFS, FileSystem targetFS) {
+        migrate(sourceFS,
+                targetFS,
+                path -> path.getFileName().toString().equals("navtree.json"));
+    }
+
+    private void migrate(FileSystem sourceFS, FileSystem targetFS, Predicate<Path> predicate) {
+        LOGGER.info("Start");
+
+        Path sourceRoot = getRoot(sourceFS);
+        Path targetRoot = getRoot(targetFS);
+
+        if (sourceRoot == null) {
+            LOGGER.warn("End: source does not exists");
+            return;
+        }
+
+        if (targetRoot == null) {
+            LOGGER.warn("End: target does not exists");
+            return;
+        }
+
+        LOGGER.info("Move from " + sourceFS.getName() + " to " + targetFS.getName());
+
+        Files.walkFileTree(sourceRoot, new SimpleFileVisitor<Path>() {
+            @Override
+            public FileVisitResult visitFile(final Path path, final BasicFileAttributes attrs) throws IOException {
+                checkNotNull("file", path);
+                checkNotNull("attrs", attrs);
+
+                Path targetPath = targetRoot.resolve(path.toString());
+
+                if (!predicate.test(path)) {
+                    LOGGER.info("Skip file " + path.toString());
+
+                } else if (ioService.exists(targetPath)) {
+                    LOGGER.warn("File " + path.toString() + " already exists on target");
+                    Files.delete(path);
+
+                } else {
+                    LOGGER.info("Move file " + path.toString());
+                    Files.copy(path, targetPath);
+                    Files.delete(path);
+                }
+
+                return FileVisitResult.CONTINUE;
+            }
+        });
+
+        LOGGER.info("End: completed");
+    }
+
+    private Path getRoot(FileSystem fileSystem) {
+        try {
+            return fileSystem.getRootDirectories().iterator().next();
+
+        } catch (Exception e) {
+            return null;
```

Comment:
Never eat exception. Can you please at least print the exception in DEBUG level (something like : LOGGER.debug("Error getting root for filesystem " + fileSystem, e);

Reply:
Done!